### PR TITLE
Visitor can see different products on a restaurants menu

### PR DESCRIPTION
**Pull request by AlexSaxena**
# Adding Product List feature
[PT-Link](https://www.pivotaltracker.com/story/show/168829452)

## PR Content
In this PR we are creating a products list for our landing page.

Includes feature and tests in step definitions.

## What we have learned 
Deeper understanding of the AUT-cycle

IMG_7396

### CODE REVIEW

File: app/controllers/landing_controller.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class LandingController < ApplicationController
```
Comment (oliverochman) [inline app/controllers/landing_controller.rb]: this should be a products controller

File: app/views/landing/index.html.haml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+%h1 Products List#index
+%p Find me in app/views/landing/index.html.haml
+%p Lasagna
```
Comment (oliverochman) [inline app/views/landing/index.html.haml]: u are hard coding these products

File: features/product_list.feature [hunk 3]
```diff
@@ -0,0 +1,12 @@
+Feature:
+    As a visitor,
+    In order to get an overview of the restaurant’s products
+    I would like to be able to see a menu
+
+Scenario: Viewing list of products on the application's landing page
+    When I'm on the landing page
```
Comment (oliverochman) [inline features/product_list.feature:14]: Please change this to:
`Given I'm on the landing page`

File: features/product_list.feature [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    Then I should see "Lasagna"
```
Comment (oliverochman) [inline features/product_list.feature:15]: Please indent this file properly

**Review by oliverochman — CHANGES_REQUESTED** [review]
This story is about presenting all of the products, but you have not done that yet.

**Review by shsteindorsson — APPROVED** [review]
Approved

**Review by shsteindorsson — APPROVED** [review]

**Review by AlexSaxena — COMMENTED** [review]
AS

**Review by beccaburns — APPROVED** [review]

**Review by lucamarial — APPROVED** [review]

**Review by miyeaier — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by miyeaier — APPROVED** [review]

**Review by miyeaier — APPROVED** [review]

**Review by miyeaier — APPROVED** [review]

**Review by lucamarial — APPROVED** [review]

**Review by lucamarial — APPROVED** [review]

**Review by lucamarial — APPROVED** [review]

**Review by beccaburns — APPROVED** [review]

**Review by miyeaier — APPROVED** [review]

**Review by shsteindorsson — APPROVED** [review]

**Review by lucamarial — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by miyeaier — APPROVED** [review]

**Review by lucamarial — APPROVED** [review]

**Review by beccaburns — APPROVED** [review]

File: app/models/product.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class Product < ApplicationRecord
+  validates :title, presence: :true
+  validates :content, presence: :true
+  validates :price, presence: :true
```
Comment (faraznaeem) [inline app/models/product.rb]: Refactor the validations

File: config/routes.rb [hunk 6]
```diff
@@ -1,2 +1,7 @@
 Rails.application.routes.draw do
-end
+
+  resources :products
+
+  root controller: :products, action: :index
+
```
Comment (faraznaeem) [inline config/routes.rb]: Please remove extra lines

**Review by ghost — APPROVED** [review]

**Review by shsteindorsson — APPROVED** [review]

File: features/product_list.feature [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    And I should see "apple pie"
+    And I should see "salad"
+    And I should see "It is good"
+    And I should see "It is tasty"
+    And I should see "It is green"
+    And I should see "100"
+    And I should see "200"
+    And I should see "300"
```
Comment (faraznaeem) [inline features/product_list.feature:23]: please group the products in order for example:
 
```
And I should see "apple pie"
And I should see "It is tasty"
And I should see "200"
And I should see "salad"
And I should see "It is green"
```
etc

File: features/step_definitions/basic_steps.rb [hunk 8]
```diff
@@ -0,0 +1,13 @@
+When("I'm on the landing page") do
+  visit root_path
+end
+
+Then("I should see {string}") do |content|
+  expect(page).to have_content content
+end
```
Comment (faraznaeem) [inline features/step_definitions/basic_steps.rb]: Move the assertion to a separate step definition file perhaps called assertions?

**Review by faraznaeem — CHANGES_REQUESTED** [review]
Good job, needs some changes until it is ready for a merge

**Review by lucamarial — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by miyeaier — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by miyeaier — APPROVED** [review]

**Review by miyeaier — APPROVED** [review]

**Review by lucamarial — APPROVED** [review]

**Review by beccaburns — APPROVED** [review]

**Review by shsteindorsson — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by lucamarial — APPROVED** [review]

**Review by shsteindorsson — APPROVED** [review]

File: config/routes.rb [hunk 9]
```diff
@@ -1,2 +1,4 @@
 Rails.application.routes.draw do
-end
+  resources :products
```
Comment (oliverochman) [inline config/routes.rb]: you need to limit what routes you have for products to what you actually use atm in the app

e.g.
`resources :products, only: [:index]`

File: features/step_definitions/assertion_steps.rb [hunk 10]
```diff
@@ -0,0 +1,5 @@
+Given("the following product exist") do |table|
```
Comment (oliverochman) [inline features/step_definitions/assertion_steps.rb]: this is not an assertion

File: features/step_definitions/basic_steps.rb [hunk 11]
```diff
@@ -0,0 +1,7 @@
+When("I'm on the landing page") do
+  visit root_path
+end
+
+Then("I should see {string}") do |content|
```
Comment (oliverochman) [inline features/step_definitions/basic_steps.rb]: this is an assertion

**Review by oliverochman — CHANGES_REQUESTED** [review]

**Review by oliverochman — APPROVED** [review]

**Pull request merged**